### ISSUE TITLE
Properly document all breaking changes

### ISSUE THREAD

**Issue by patrick-stalph** [oxyplot/oxyplot#2105]
All oxyplot _minor_ version upgrades so far have broken our builds in unexpected ways. I was expecting the same to happen with 2.2 and I was right. In particular I would like to understand why `LinearAxis.CalculateActualInterval(double, double)` is not available anymore (compiler error: no suitable method to override) and how to change my code . Consequently I was reading the [changes here](https://github.com/oxyplot/oxyplot/releases/tag/v2.2.0) but couldn't find a hint.

You should consider documenting your breaking API changes in more detail by programmatically comparing API changes. Andrew Lock has written a nice blog post with two different approaches for library developers:
https://andrewlock.net/preventing-breaking-changes-in-public-apis-with-publicapigenerator/
Such a mechanism would make you aware of breaking changes and allow you to better document why the API changed and what developers should use instead.

**Comment by VisualMelon** [comment]
Thanks for raising this general issue, and highlighting the specific breaking change.

The loss of that overload looks like lazy design/a mistake on my part. Relevant PR: https://github.com/oxyplot/oxyplot/pull/1712/files#diff-cf8d2faf9e8ba3fcd2b20cce299a0a0557f682542cddcc57a444673a9d9c829cR1823-R1830. I would be happy to put the old overloads back in if it's a significant issue, though would be interested to know why you are calling the method: my instinct is it will probably be brittle, and probably shouldn't have been part of the public API in the first place. Could do with throwing more exceptions in there as well.

As a solution for the future, adding Microsoft.CodeAnalysis.PublicApiAnalyzers with Shipped/Unshipped listings as discussed in the blog seems reasonable to me: so long as we remember to move everything to shipped with each release then it should hopefully give us a sensible diff and should put negligible burden on contributors. The problem will still be that writing good release notes is time-consuming, but this should at least make it easier to work out what needs to be written - as you say - and make it easier for people to provide feedback prior to a release.

Example for core lib: https://github.com/VisualMelon/oxyplot/commit/44dc2ad5c0653588befb91d464b99406bcd555c2

**Comment by patrick-stalph** [comment]
I have a few custom `LinearAxis` implementations specific to our scenarios (e.g. work with metric data under the hood, but show international feet in the plot or whatever unit the user selected in the UI). Often it is difficult to find out how to solve my tasks with OxyPlot. Usually I inspect the OxyPlot source to find solutions for my offroad requirements.

For my particular compiler error on `CalculateActualInterval` I found that new arguments were added to this method and I can simply forward those.
